fmn/delivery/backends/mail.py

```diff
@@ -0,0 +1,131 @@
+from .base import BaseBackend
+
+from kitchen.text.converters import to_bytes
+
+import email
+import logging
+
+from twisted.internet import defer
+from twisted.mail import smtp
+
+from fmn.lib import models
+from fmn.util import get_fas_email
+
+_log = logging.getLogger(__name__)
+
+
+CONFIRMATION_TEMPLATE = u"""
+{username} has requested that notifications be sent to this email address
+* To accept, visit this address:
+  {acceptance_url}
+* Or, to reject you can visit this address:
+  {rejection_url}
+Alternatively, you can ignore this.  This is an automated message, please
+email {support_email} if you have any concerns/issues/abuse.
+"""
+
+reason = u"""
+You received this message due to your preference settings at
+{base_url}{user}/email/{filter_id}
+"""
+
+
+class EmailBackend(BaseBackend):
+    __context_name__ = "email"
+
+    def __init__(self, *args, **kwargs):
+        super(EmailBackend, self).__init__(*args, **kwargs)
+        self.mailserver = self.config['fmn.email.mailserver']
+        if ':' in self.mailserver:
+            self.host, self.port = self.mailserver.split(':')
+        else:
+            self.host, self.port = self.mailserver, 25
+        self.port = int(self.port)
+        self.from_address = self.config['fmn.email.from_address']
+
+    @defer.inlineCallbacks
+    def deliver(self, formatted_message, recipient, raw_fedmsg):
+        """
+        Deliver a message to the recipient.
+
+        Args:
+            formatted_message (str): The formatted message that is ready for delivery
+                to the user. It has been formatted according to the user's preferences.
+            recipient (dict): The recipient of the message.
+            raw_fedmsg (dict): The original fedmsg that was used to produce the formatted
+                message.
```

Comment:
It would be good to document here that this is a generator and yields the return value of sendmail(), whatever that is.

Reply:
Also, why is this a generator? Seems a bit strange.

Reply:
The reason it's a generator is because it's using Twisted's inlineCallback decorator. When something returns a deferred (same thing as a future, promise, etc) you yield it and the the function will continue when the deferred fires. This lets you do things like try/except rather than defining callbacks for errors.